evm/chains/chain.py

```diff
@@ -279,21 +275,21 @@ def get_chain_at_block_parent(self, block):
             ))
 
         init_header = self.create_header_from_parent(parent_header)
-        return type(self)(self.db, init_header)
+        return type(self)(self.chaindb.raw_db, init_header)
```

Comment:
Looking at this line as well as [this line from the `from_genesis` method](https://github.com/pipermerriam/py-evm/pull/114/files#diff-38ff1253a2c8ee641d11e159b1584c5fR208) makes me think that we might want to just pass in an instantiated `ChainDB` object rather than the underlying database backend.

Reply:
Yeah, makes sense, and then I can get rid of Chain.chaindb_class